### DISCUSSION TITLE
Question about vector files, .txt documents, and navigation addresses...

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#56443] [Help] [answered]
### Summary

So I am working with a Langchain code that I am converting from plain Node.js to Next.js..

Code is below..

Here are some questions:

1. Can Next.js work with Faiss vector store files?   I ask because I am working with Faiss, but for some reason from the Langchain documentation it is marked as "node-only"?...  link here: https://js.langchain.com/docs/modules/data_connection/vectorstores/integrations/faiss
^^  If this vector store doesn't work in Next.js....  can anybody recommend a vector store that can work with Next.js?

2.  I have a txt file that I need to feed it to the vector store, and so it would be return to as a vector (I think) to work with Langchain.....  Can I work with txt files in Next.js?  or do I need to use a different format?

3. I actually got this piece of code working from Node.js to Next.js... meaning it isn't giving me errors when I run node `pages/api/indexes.js `  ... however it is doing NOTHING..   it is suppose to create a couple files and save them, but I just see nothing happening, not even errors...... I wonder..... What would be the correct path to give to point to a file, and then point to create documents in a given location?... I think it is possible the problem might be in the path given.

Code is here:

```

const { OpenAIEmbeddings } = require("langchain/embeddings/openai");
const { FaissStore } = require("langchain/vectorstores/faiss");
const { CharacterTextSplitter } = require("langchain/text_splitter");
const { TextLoader } = require("langchain/document_loaders/fs/text");

const indexes = async (req, res) => {
  const loader = new TextLoader("../../pages/api/restaurant.txt");
  console.log(loader);
  const docs = await loader.load();
  const splitter = new CharacterTextSplitter({
    chunkSize: 200,
    chunkOverlap: 50,
  });
  const documents = await splitter.splitDocuments(docs);
  console.log(documents);

  const embeddings = new OpenAIEmbeddings();
  const vectorStore = await FaissStore.fromDocuments(documents, embeddings);
  await vectorStore.save("./");

  res.status(200).json({ ok: ok });
};

module.exports = { indexes };

```

### Additional information

```js
https://github.com/Elindo586/nextprojectd/blob/main/pages/api/indexes.js
```

**Comment by icyJoseph** [comment]
I guess you are doing something like `node pages/api/index.js`, but why would you expect anything to happen? You never call `indexes`. 

The handler is only called if you make a REST request to `pages/api/indexes` 

However, in Vercel deployments, API routes can only have temporary storage, https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage, until the instance is killed. Then when a new request arrives, it is awaken clean once again. 

- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002

**Reply by Elindo586** [reply]
Yes, you are correct... the command I am calling is `node pages/api/indexes.js`  <-- this command just doesn't show anything.


Is the `pages/api ` able to get documents through paths like  `const loader = new TextLoader("../../pages/api/restaurant.txt");`  .. and then save a return document to a different path like ` await vectorStore.save("./");`  ??

I might be having a problem typing the correct path to get to the  `restaurant.txt `document...  which is saved under `pages/api`

If the `pages/api`  is not able to get and save documents this way, would there be an alternative way to do it?

Right now I am doing this:

1. Manually create a .txt document with information and right now saved under `pages/api/restaurant.txt`
2. From `pages/api/indexes.js`  call and load the `restaurant.txt` to be processed with the Langchain Faiss Vector Store.
3. Once it is processed by the vector store, the store will return 2 different documents to be saved back into my computer.. I could save it under `pages/api `??? or some other location..
4. Once the documents are saved in my local computer, then I can call them from a different file to use that information.


^^^  This is kind of the way it is used in plain Node.js, but the documents are being use from the root instead of putting them in a file...  as an addition to my doubts is that I am never sure if I am typing to correct paths to get the and save the files properly.

**Comment by Elindo586** [comment, marked as answer]
Ok...

The problem was the I needed to npm the faiss-node...

I think the other problem was that I couldn't do a  `node pages/api/indexes.js `command... it wasn't giving me any errors...  So I ended up integrating this piece of code with another api page that runs when I supply a text to it.... 

The bot still not working as I want, but I still need to add more code to it...
